lib/components_test.go

```diff
@@ -0,0 +1,39 @@
+package lib
```

Comment:
Having them in the same package means the test files get included in the final build, right? Not a _huge_ problem, but this could make the resulting binary larger than it needs to be.